File: src/main/java/com/ggardet/hospital/spi/dto/TreatmentDTO.java
package com.ggardet.hospital.spi.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

import java.util.UUID;

public record TreatmentDTO(
        UUID id,

        @NotBlank String name,

        String description,

        @Positive Integer durationMinutes,

        String protocol
) {
}

File: src/main/kotlin/com/ggardet/hospital/spi/dto/TreatmentDTO.kt
package com.ggardet.hospital.spi.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import java.util.*

data class TreatmentDTO(
    val id: UUID? = null,

    @field:NotBlank 
    val name: String,

    val description: String? = null,

    @field:Positive 
    val durationMinutes: Int? = null,

    val protocol: String? = null
)